classes/class3/homework/routes/cats.js

```diff
@@ -0,0 +1,90 @@
+var path = require('path');
+var Cat = require(path.join(__dirname, '../models/cat'));
+
+var colors = ['yellow',
+              'green',
+              'red',
+              'black',
+              'blue',
+              'orange',
+              'tabby',
+              'NaN',
+              'brown',
+              'teal',
+              'lemmony yellow',
+              'grey',
+              'gray',
+              'maroon',
+              'macaroon'];
+
+var names = ['George',
+             'Malyssa',
+             'Frederick',
+             'Johan',
+             'Suzy',
+             'Mary',
+             'Skags',
+             'Robert',
+             'Tinkerbell',
+             'Mr. Whiskers',
+             'King Fluffybutt']
+
+var cats = {};
+
+function all(req, res) {
+  var colorFilter;
+  Cat.find({colors: colorFilter}, function(err, cats) {
+    res.render('cats', {cats: cats});
+  });
+}
+
+function create(req, res) {
+  var color = colors[Math.floor(Math.random() * colors.length)];
+  var name = names[Math.floor(Math.random() * names.length)];
+  var age = Math.floor(Math.random() * name.length);
+  var catObject = {
+    name: name,
+    age: age,
+    colors: color
+  };
+  var newCat = new Cat(catObj);
+  newCat.save(function(err) {
+    if (err) {
+      console.log('Cat object couldn\'t be saved');
+    } else {
+      res.render('cats', [catObj]);
+    }
+  });
+}
+
+function list(req, res) {
+  var colorFilter;
+  if (req.params.color) {
+    colorFilter = req.params.color.toLowerCase();
+  } else {
+    colorFilter = '';
+  }
+  Cat.find({colors: colorFilter}).sort({age: -1}).exec(function(err, cats) {
```

Comment:
Despite the `.sort()` your cats are managing to not be sorted by age.

[Example](https://www.dropbox.com/s/symfwmlatlz9ihs/Screenshot%202015-02-03%2020.13.01.png?dl=0)

You've got some interesting things going on in your app here.